### PR TITLE
Retrieve telemetry send status from server component

### LINKED ISSUE #7571: Create admin data table and store the telemetry status
The installation-telemetry is an opt-out service where the status will be stored in a database. A database is use to allow for the admin users to opt in or out without having to reconfigure the application at cluster-level.

This will likely be more of an admin data table where the telemetry status is one part. It may require key/value pair support as well.

As a first iteration, suggest calling this `d_b_installation_admin` with a boolean type for `telemetry`, defaulting to `true`

## Potential problems:

- ~we use NodeJS TypeORM to generate the migration scripts. This will be (I believe) the first Golang component that will access the MySQL database so the established pattern will need thinking about. It will also need NodeJS support as the dashboard will have to access this.~ Having talked through the problem with @corneliusludmann, we've suggested that creating an API endpoint in `server` and having the `installation-telemetry` component call it would be the best solution to this. This requires an API endpoint (and appropriate authentication) inside `server`

### LINKED ISSUE #7585: Create an installation-admin endpoint inside server component
Follows on from #7571

In order to maintain consistency in the database, an installation-admin endpoint should be created in the `server` to provide read-only access to the data. This will be used for the `installation-telemetry` component to read whether it should send telemetry data to SegmentIO.

This endpoint may need appropriate authentication to ensure that this data is not used nefariously. Whilst the initial "can I send telemetry data" may be low-risk if shared, in future this could have more sensitive data inside

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Creates the installation admin controller in the `server` component and associated database. This database works on the basis that there is only one record created per installation - in future, this will be the backing for the installation's admin panel. At the moment, this only has one record (`sendTelemetry`), but will have more records in future.

The `sendTelemetry` boolean currently defaults to `false` and will do until the admin panel exists then it will change to default `true`.

It amends the Installer to cater for these changes. It allows the `gitpod-telemetry` job to access the `server` component. It also removed the `GITPOD_DOMAIN_HASH` and uses the randomly generated ID for installation identification.

## Related Issue(s)

Fixes #7571
Fixes #7585 

## How to test

Deploy via werft and check that you can access `GET:/installation-admin/data` when port forwarding to the server component.

## Release Notes

```release-note
[server]: Create installation admin controller
```

### CODE REVIEW

**Comment by roboquat** [comment]
@MrSimonEmms: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **alextugarev** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @alextugarev` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **alextugarev** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @alextugarev` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **jldec** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jldec` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **laushinka** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @laushinka` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** and additionally assign **laushinka** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @laushinka` in a comment when ready.

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/gitpod-protocol/src/teams-projects-protocol.ts
Comment (corneliusludmann) [inline components/gitpod-protocol/src/teams-projects-protocol.ts:120]: I wonder if the file `teams-projects-protocol.ts` is the proper place. What do you think of creating a dedicated file for this (`installation-admin-protocol.ts` or something similar)?

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (corneliusludmann) [inline components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:26]: Shouldn't it be `true` because it's “opt-out”?

```suggestion
            sendTelemetry: true,
```

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (corneliusludmann) [inline components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:25]: Can/should we use this ID as installation ID and replace the domain hash that we currently use as installation ID? Once generated and saved to the DB, this ID shouldn't change anymore, right? What do you think?

File: components/server/src/installation-admin/installation-admin-controller.ts
Comment (corneliusludmann) [inline components/server/src/installation-admin/installation-admin-controller.ts:21]: I feel a bit uncomfortable with simply sending the whole record over this unprotected endpoint. If someone adds additional properties to the database record, the person needs to have in mind that it will be exposed here. Maybe we filter the data at this point and expose only the data that is needed by the telemetry component? What do you think? Are my concerns exaggerated?

We could also create a `getTelemetryData(...)` function instead and let `InstallationAdminDB` filter the stuff for us.

**Review by corneliusludmann — COMMENTED** [review]
Thanks for this draft. :+1: 

Added some random comments.

File: components/gitpod-protocol/src/teams-projects-protocol.ts
Comment (MrSimonEmms) [reply components/gitpod-protocol/src/teams-projects-protocol.ts:120]: Yeah, it probably isn't the correct place at all. I think when I was writing it, I misunderstood that file as "HERE ARE ALL THE THINGS"

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (MrSimonEmms) [reply components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:26]: It will be `true` when the dashboard happens. See description:

> The sendTelemetry boolean currently defaults to `false` and will do until the admin panel exists then it will change to default `true`.

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (MrSimonEmms) [reply components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:25]: It's not impossible for the installation ID to change - if someone changes their domain and/or changes their pepper (see #7538), this will change the installation ID. Personally, I feel it's better to decouple the installation ID and the DB record ID

File: components/server/src/installation-admin/installation-admin-controller.ts
Comment (MrSimonEmms) [reply components/server/src/installation-admin/installation-admin-controller.ts:21]: Yeah, I'm fine with that.

Concerns are probably a little exaggerated, but I'd rather be safe than sorry so I'll change to `getTelemetryData`

File: components/gitpod-db/src/typeorm/installation-admin-db-impl.ts
Comment (corneliusludmann) [reply components/gitpod-db/src/typeorm/installation-admin-db-impl.ts:26]: Ah, makes sense. :+1:

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl**, **mrsimonemms** after the PR has been reviewed.

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/gitpod-db/src/typeorm/migration/1642422506330-InstallationAdminTable.ts
Comment (geropl) [inline components/gitpod-db/src/typeorm/migration/1642422506330-InstallationAdminTable.ts:13]: We usually guard with `if (!(await tableExists(...))) {...}` to ensure we can safely re-execute.

File: components/gitpod-db/src/typeorm/entity/db-installation-admin.ts
Comment (geropl) [inline components/gitpod-db/src/typeorm/entity/db-installation-admin.ts:17]: As mentioned by @JanKoehnlein yesterday we should thrive for as little "potential change surface" as possible in `gitpod-db` and `server` to avoid the cross-team overhead as much as possible.

I suggest a colum like this:
```
@Column("simple-json")
settings: InstallationAdminSettings;
```

File: components/gitpod-protocol/src/installation-admin-protocol.ts
Comment (geropl) [inline components/gitpod-protocol/src/installation-admin-protocol.ts:13]: If we made this `sendTelemetry?: boolean,` we could avoid settings an explicit default (as long as it's `false`, that is).

Also, :+1: for using an interface (if you forsee additional fields):
```
export interface Telemetry {
    enabled: boolean,
    // userStatistics: boolean,
    // traceEvenets: boolean,
}
```

File: components/server/src/server.ts
Comment (geropl) [inline components/server/src/server.ts:288]: @MrSimonEmms This is publicly available at `gitpod.io/api/installation-admin`. Either we use a custom express app at another port (see this.monitoringApp) - or use the DB as interface in the first place.

/cc @JanKoehnlein

File: components/server/src/installation-admin/installation-admin-controller.ts
Comment (geropl) [reply components/server/src/installation-admin/installation-admin-controller.ts:21]: I don't think we should start filtering here. Instead, I propose we either:
 - make the entire endpoint non-public (see comment in server.ts)
 - use the DB as interface in the first place (not sure whether this is actually a good idea)

If we keep the endpoint, using `res.status(200).json(data);` will set correct content type header as well. :+1:

File: components/gitpod-protocol/src/installation-admin-protocol.ts
Comment (MrSimonEmms) [reply components/gitpod-protocol/src/installation-admin-protocol.ts:13]: My expectation is that `sendTelemetry` will always be populated - if the record doesn't exist in the DB, it's created for us and the parameter is set.

Unless you strongly favour an optional flag, I'd suggest keeping it as required

File: components/server/src/server.ts
Comment (MrSimonEmms) [reply components/server/src/server.ts:288]: @geropl personally, creating an additional port just seems messy. How would you feel about an auto-generated HTTP Basic auth adding to the endpoint?

File: components/gitpod-protocol/src/installation-admin-protocol.ts
Comment (MrSimonEmms) [reply components/gitpod-protocol/src/installation-admin-protocol.ts:13]: There definitely will be additional fields too

Comment (geropl) [reply components/gitpod-protocol/src/installation-admin-protocol.ts:13]: > My expectation is that sendTelemetry will always be populated - if the record doesn't exist in the DB

Got that. My point here ws that we should try hard to minimize the code changes required here. ATM one could argue that `server`/`gitpod-db` don't need to now _anything_ about the internal structure of the `InstallationAdminSettings` shape.
But maybe this is just starting yesterdays discussion all over again.

File: components/server/src/server.ts
Comment (geropl) [reply components/server/src/server.ts:288]: > creating an additional port just seems messy

@MrSimonEmms  Definitely agree. But worse then an outward facing endpoint? Dunno. I'm just not super happy with this integration point in general.
But yesterdays discussions, again. Let's start out with this and look for alternative approaches! :+1: :pray: 

> How would you feel about an auto-generated HTTP Basic auth adding to the endpoint?

If the token/password is stable (does not change when upgrading a given installation) fine with me.

File: components/gitpod-protocol/src/installation-admin-protocol.ts
Comment (MrSimonEmms) [reply components/gitpod-protocol/src/installation-admin-protocol.ts:13]: I guess I'm in your hands as you're the owners of this component. Realistically, I'm happy to do anything you want - I don't even mind if you wanted to do the changes in your own team although I've proposed doing them in my team as I know how much work you have on.

So long as we have a generated ID that remains the same per-installation, and a `sendTelemetry` boolean that we can update, I don't mind how you want me to do it.

Comment (geropl) [reply components/gitpod-protocol/src/installation-admin-protocol.ts:13]: > So long as we have a generated ID that remains the same per-installation, and a sendTelemetry boolean that we can update, I don't mind how you want me to do it.

Alright. Current code LGTM :slightly_smiling_face:

**Comment by MrSimonEmms** [comment]
This one's been a little bit complex, but I've done it as per the discussions with @geropl and @corneliusludmann

I've created an additional "installation admin" express app which is hosted on port `9000`, purely for the installation admin controller. This is to avoid this endpoint being available over the main `/api` endpoint, similar to how the monitoring endpoint works.

I've updated the Installer & telemetry component to work with that and have tested it in my self-hosted cluster

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl**, **mrsimonemms** after the PR has been reviewed.

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl**, **mrsimonemms** after the PR has been reviewed.

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by corneliusludmann** [comment]
Looks good from my side. Would like to leave the final `/lgtm` to Gero because of the server/DB changes.

/approve

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1018338659" title="Approved">corneliusludmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- ~~[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)~~ [corneliusludmann]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [corneliusludmann]
- **[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by corneliusludmann** [comment]
@MrSimonEmms This is ready for review, right? Could we remove the hold label?

**Comment by MrSimonEmms** [comment]
/unhold

@corneliusludmann yes - left on my mistake

**Review by geropl — APPROVED** [review]
LGTM!

(sorry for the delay :pray: )

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: a207956f456e8c67fb7e087db670e5fd87a53c72</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1018338659" title="Approved">corneliusludmann</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7591#pullrequestreview-861960414" title="Approved">geropl</a>*

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)~~ [corneliusludmann]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [corneliusludmann]
- ~~[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
@MrSimonEmms: you cannot LGTM your own PR.

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1021087464):

>/lgtm
>/approve


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1018338659" title="Approved">corneliusludmann</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7591#pullrequestreview-861960414" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1021087464" title="Approved">MrSimonEmms</a>*

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)~~ [MrSimonEmms,corneliusludmann]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [MrSimonEmms,corneliusludmann]
- ~~[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 7889540a246a1bfaaf1ba8695e6f4a7ca576c83a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1018338659" title="Approved">corneliusludmann</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1021090067" title="LGTM">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7591#issuecomment-1021087464" title="Approved">MrSimonEmms</a>*

Associated issue: *#7571*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]
- ~~[components/installation-telemetry/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/installation-telemetry/OWNERS)~~ [MrSimonEmms,corneliusludmann]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]
- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [MrSimonEmms,corneliusludmann]
- ~~[installer/pkg/components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
